### DISCUSSION TITLE
Interface with relational template

### DISCUSSION THREAD

**Discussion by CapitaineToinon** [directus/directus#14197] [Archive]
Hey! I'm trying to make an extension that has a template field to format some text based on relational properties and I'm trying to get relational fields to work. I've looked at a bunch of code examples (such as https://github.com/dimitrov-adrian/directus-extension-wpslug-interface) and the source code directly to figure out how to get it to work but I can't get it to work!

I found [this message](https://github.com/directus/directus/discussions/11255#discussioncomment-2039098) that (I think) describe the same problem but with no solution

Basically I'm trying to have something like `{{ parent.slug }}/{{ slug }}` to be compiled properly in my interface. However by using the code `const values = inject('values', ref<Record<string, any>>({}))` inside my interface, the values won't have relational data, only the id (so in the example, I will have `{ parent: 1, ... }` so it obviously doesn't work.

I know we can do nested templates in the dashboard app so I looked at the code and I found the function `useTemplateData` in here https://github.com/directus/directus/blob/8f00e37daf421c0378544a2a953a8549f043875e/app/src/composables/use-template-data.ts#L12 and I'm not sure if that would work. It seems I can access that function anyway. I would love some help here.

Maybe I'm approaching the problem the wrong way. I'm trying to create a tree view of pages and the idea would be to let users created nested complicated pages and have the deep slug (url) to be automatically generated base on the parent slug. 

So the user could create the page with title "My musics" and I'm using a slug libary to get the url `/my-musics`. Then the user can create a new page with title `My first song` with the former as a parent page and the idea is to generate a slug as `/my-musics/my-first-song`. 

I know given a node, I could reconstruct the full slug manually but I'm trying to access it the other way around, given the full slug I want to be able to retrieve the correct page. 

Finally, I am aware that an interface will only work through the app and not the api but for now that's not an issue.

**Comment by rijkvanzanten** [comment]
I _think_ that you'll have to manually retrieve the correct data based on the primary keys you have available through your earlier attempt:

> Basically I'm trying to have something like {{ parent.slug }}/{{ slug }} to be compiled properly in my interface. However by using the code const values = inject('values', ref<Record<string, any>>({})) inside my interface, the values won't have relational data, only the id (so in the example, I will have { parent: 1, ... } so it obviously doesn't work.

The app won't magically have all the data available, and if you look at that `useTemplateData` utility function, that also "just" fetches the items fresh based on the fields used in the template. You should be able to use that function as inspiration for a starting point. You can access the `api` instance through the `useApi` composable from the `extensions-sdk`:


```
import { useApi } from '@directus/extensions-sdk`;

const api = useApi();
```
